draft-dekater-scion-controlplane.md

```diff
@@ -1297,6 +1309,64 @@ In comparison to these time scales, clock offsets in the order of minutes are im
+If the same AS has 1000 child links, the propagation of the beacons will require signing one new AS entry for each of the propagated PCBs for each link, that is at most 50000 signatures per propagation interval.
+The total bandwidth for the propagation of these PCBs for all 1000 child links would, again very roughly, be around 25MB/s.
+All of these are managable with even modest consumer hardware.
+
+
+On a cold start of the network, path segments to each AS are discovered after a number of propagation steps proportional to the longest path. As mentioned, the longest path is typically not long. With a 5 second propagation period and a generous longest path of length 10, all path segments are discovered after 25 seconds on average.
+
+When a new parent-child link is added to the network, the parent AS will propagate the available PCBs in the next propagation interval. If the AS on the child side of the new link is a leaf AS, path discovery is thus complete after one single propagation interval. Otherwise, child ASes at distance D below the new link, learn of the new link after D further propagation steps.
+
+### Inter-ISD Beaconing
+In the inter-ISD core beaconing, PCBs are propagated omnidirectionally along core links. Each AS discovers path segments from itself to any other core AS.
+The number of distinct paths through the core network is typically very large. To keep the overhead managable, at most 5 path segments to every destination AS are discovered, and the propagation frequency is slower than in the intra-ISD beaconing.
+
+Without making strong assumptions on the topology of the core network, we can assume that shortest paths through real world, internet-like networks are relatively short; for example, the Barabási-Albert random graph model predicts a diameter below log(N)/log(log(N)) for a network with N nodes {{BollRio-2000}}.
+We cannot assume that the selected PCBs are strictly shortest paths through the network, but it's reasonable to assume that they will not be very much longer than the shortest paths either. It seems safe to assume that the PCBs are not longer than 10 hops on average for realistic networks.
+<-- TODO find something about _average_ shortest path. Probably can say something smaller than 10 -->
+
+With N the number of participating core ASes, an AS receives up to 5 * N PCBs per propagation interval per core link interface.
+For highly connected ASes, the number of PCBs received thus becomes rather large. In a network of 1000 ASes, an AS with 200 core links receives up to one million PCBs per propagation interval.
+This corresponds to roughly 150 thousand signature validations per second. This is just about the achievable throughput on a single core of a present day small server or desktop machine.
+In terms of bandwidth, this corresponds to very roughly 40MB/s.
```

Comment:
Given your previous estimate of 250bytes per AS entry and core segments of ~10 ASes, I compute:
200000 * 10 * 250 = 500M/s. Even using your 150K/s figure, I still get 375M/s. I missed something?

Reply:
Ack. I had missed something.

Reply:
I've changed the numbers for the example; I noticed that length 10 is much longer than what we say above with the log n/loglog n average path length.
For 1000 ASes, the log n/loglog n is around 3.5. Because we say that we have paths that are longer than the shortest path, I use a generous 6 -- because this gives nice numbers.
Because the numbers coincidentally ended up confusingly similar to the intra-ISD case I've increased the number of core links from 200 to 300.

So:
5 * 300 * 1000 = 1.5 million PCBs
5 * 300 * 1000 * 6 / 60 = 150k AS entries per second
5 * 300 * 1000 * 6 * 250 / 60 = 37.5MB/s